test/fishjam/cluster/api_test.exs

```diff
@@ -0,0 +1,248 @@
+
+    assert_room_count_on_fishjam(fishjam_instance1, 1)
+
+    assert_room_count_in_cluster(node1, 1)
+    assert_room_count_in_cluster(node2, 1)
+
+    response_body2 = add_room(node1)
+
+    fishjam_instance2 = get_fishjam_address(response_body2)
+
+    assert_room_count_on_fishjam(fishjam_instance2, 1)
+
+    assert_room_count_on_fishjam(node1, 1)
+    assert_room_count_on_fishjam(node2, 1)
+
+    assert_room_count_in_cluster(node1, 2)
+    assert_room_count_in_cluster(node2, 2)
+
+    room_id = response_body1 |> Jason.decode!() |> get_in(["data", "room", "id"])
+
+    delete_room(fishjam_instance1, room_id)
+
+    assert_room_count_on_fishjam(fishjam_instance1, 0)
+    assert_room_count_on_fishjam(fishjam_instance2, 1)
+
+    assert_room_count_in_cluster(node1, 1)
+    assert_room_count_in_cluster(node2, 1)
+
+    response_body3 = add_room(node1)
+    fishjam_instance3 = get_fishjam_address(response_body3)
+    assert_room_count_on_fishjam(fishjam_instance3, 1)
+
+    assert_room_count_on_fishjam(node1, 1)
+    assert_room_count_on_fishjam(node2, 1)
+
+    assert_room_count_in_cluster(node1, 2)
+    assert_room_count_in_cluster(node2, 2)
+  end
+
+  @tag timeout: @max_test_duration
+  test "request routing within a cluster, using room_id", %{nodes: [node1, node2]} do
+    response_body1 = add_room(node1)
+
+    room_node = get_fishjam_address(response_body1)
+    other_node = if room_node == node1, do: node2, else: node1
+
+    assert_room_count_on_fishjam(room_node, 1)
+    assert_room_count_on_fishjam(other_node, 0)
+
+    assert_room_count_in_cluster(room_node, 1)
+    assert_room_count_in_cluster(other_node, 1)
+
+    room_id = response_body1 |> Jason.decode!() |> get_in(["data", "room", "id"])
+
+    _response_body2 = add_peer(other_node, room_id)
+
+    assert_peer_count_in_room(room_node, room_id, 1)
+    assert_peer_count_in_room(other_node, room_id, 1)
+
+    _response_body3 = delete_room(other_node, room_id)
+
+    assert_room_count_on_fishjam(room_node, 0)
+    assert_room_count_on_fishjam(other_node, 0)
+
+    assert_room_count_in_cluster(room_node, 0)
+    assert_room_count_in_cluster(other_node, 0)
+  end
+
+  @tag timeout: @max_test_duration
+  test "request routing + explicit forwarding of HLS retrieve content requests", %{
+    nodes: [node1, node2]
+  } do
+    response_body1 = add_room(node1)
+
+    room_node = get_fishjam_address(response_body1)
+    other_node = if room_node == node1, do: node2, else: node1
+
+    assert_room_count_on_fishjam(room_node, 1)
+    assert_room_count_on_fishjam(other_node, 0)
+
+    assert_room_count_in_cluster(room_node, 1)
+    assert_room_count_in_cluster(other_node, 1)
+
+    room_id = response_body1 |> Jason.decode!() |> get_in(["data", "room", "id"])
+
+    _response_body2 = add_hls_component(other_node, room_id)
+
+    _response_body3 = add_file_component(other_node, room_id)
+
+    # Wait a while for segments and manifest to get created
+    Process.sleep(20_000)
```

Comment:
![image](https://github.com/fishjam-dev/fishjam/assets/56085570/a81cd5ce-1143-4913-8cbd-7812df695cd6)

Reply:
Any other ideas are welcome XD

Reply:
I think we could subscribe on server notifications and await on `HlsPlayable`; this way, it should be a little bit shorter.